src/http/api/webhook.ts

```diff
@@ -51,22 +53,19 @@ export function createPostPublishedWebhookHandler(
                 (await ctx.req.json()) as unknown,
             ).post.current;
         } catch (err) {
-            return new Response(JSON.stringify({}), { status: 400 });
+            if (err instanceof Error) {
+                return BadRequest(`Could not parse payload: ${err.message}`);
+            }
+            return BadRequest('Could not parse payload');
         }
 
         const account = await accountRepository.getBySite(ctx.get('site'));
 
-        let post: Post | null = null;
-        try {
-            post = Post.createArticleFromGhostPost(account, data);
-            await postRepository.save(post);
-        } catch (err) {
-            ctx.get('logger').error('Failed to store post: {error}', {
-                error: err,
-            });
-        }
+        const post = Post.createArticleFromGhostPost(account, data);
+
+        await postRepository.save(post);
 
-        return new Response(JSON.stringify({}), {
+        return new Response(JSON.stringify(postToDTO(post)), {
```

Comment:
This allows us to remove the outbox checks from the tests and get access to the id